service.kn.switchtimer/addtimer.py

```diff
@@ -0,0 +1,17 @@
+import xbmc
+import handler
+
+if __name__ ==  '__main__':
+
+    handler.notifyLog('Parameter handler called: add Timer')
+    try:
+        args = {'channel':xbmc.getInfoLabel('ListItem.ChannelName'),
```

Comment:
Although I am not sure if it is possible to get the channelname that way....
If not, then your way is fine enough.